### PR TITLE
[tests-only][full-ci] Added API tests to check file versions of shared file from project space by a sharee

### PR DESCRIPTION
**Pull request by ghost**
## Description

Added API tests for checking file versions of shared file from a space (file shared with different role)

```feature
Scenario Outline: check the file versions of a shared file
    Examples:
      | role   | 
      | editor | 
      | viewer |
      | all    |
```

## Related Issue

- Part of : https://github.com/owncloud/ocis/issues/7824

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Feature: Share a file or folder that is inside a space and check the file version
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:1]:
```suggestion
Feature: checking file versions using file id
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  Scenario: file version
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:3]:
```suggestion
  Scenario: check the file versions of a shared file
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    And using spaces DAV path                                                                                                      
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API                                  
+    And user "Alice" has created a space "version check" with the default quota using the Graph API                                
+    And user "Alice" has uploaded a file inside space "version check" with content "hello world version 1" to "text.txt"
+    And we save it into "FILEID"          
+    And user "Alice" has uploaded a file inside space "version check" with content "hello world version 1.1" to "text.txt"         
+    When user "Alice" creates a share inside of space "version check" with settings:                                               
+      | path      | text.txt |
+      | shareWith | Brian    |
+      | role      | editor   |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:17]: make it Given step

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    When user "Alice" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
+    Then the HTTP status code should be "207"
+    Then the number of versions should be "1"
+    When user "Brian" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:21]:
```suggestion
    When user "Brian" tries to get the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```

File: tests/acceptance/features/bootstrap/FilesVersionsContext.php [hunk 5]
```diff
@@ -101,6 +101,28 @@ public function getFileVersions(
 		);
 	}
 
+	/**
+	 * @When user :user gets the number of versions of file :resource using file-id path :endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FilesVersionsContext.php:105]:
```suggestion
	 * @When user :user gets the number of versions of file :resource using file-id path :endpoint
	 @When user :user tries to get the number of versions of file :resource using file-id path :endpoint
```

File: tests/acceptance/features/bootstrap/FilesVersionsContext.php [hunk 6]
```diff
@@ -101,6 +101,28 @@ public function getFileVersions(
...
+	public function userGetsTheNumberOfVersionsOfFileOfTheSpace(string $user, string $endpoint):void {
+		$this->featureContext->setResponse(
+			$this->featureContext->makeDavRequest(
+				$user,
+				"PROPFIND",
+				$endpoint,
+				null,
+				null,
+				null,
+				"2"
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/FilesVersionsContext.php:121]: why static `2` would it not work with dav version `1` or spaces?

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  So that other users can access the file
+
+  Scenario:  check the file versions of a shared file
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+    And using spaces DAV path                                                                                                      
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API                                  
+    And user "Alice" has created a space "version check" with the default quota using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:13]: let's not use action as a name for something
```suggestion
    And user "Alice" has created a space "Project101" with the default quota using the Graph API                                
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    And user "Alice" has created a space "Project1" with the default quota using the Graph API                                
+    And user "Alice" has uploaded a file inside space "Project1" with content "hello world version 1" to "text.txt"
+    And we save it into "FILEID"          
+    And user "Alice" has uploaded a file inside space "Project1" with content "hello world version 1.1" to "text.txt"
+    And user "Alice" has created a share inside of space "Project1" with settings:
+      | path      | text.txt |
+      | shareWith | Brian    |
+      | role      | editor   |
+    And using new DAV path
+    When user "Alice" tries to get the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:22]:
```suggestion
    When user "Alice" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 9]
```diff
@@ -0,0 +1,26 @@
+Feature: checking file versions using file id
+  As a user
+  I want to share file outside of the space
+  So that other users can access the file
+
+  Scenario:  check the file versions of a shared file
```
Comment (SagarGi) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:6]:
```suggestion
  Scenario: check the file versions of a shared file
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by v-scharf — APPROVED** [review]
Could you please check what behavior if you share space with a file to user. 
Can he see the version of the file and what role should he have?
do we have test for that?

of course in different PR

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 10]
```diff
@@ -0,0 +1,27 @@
+Feature: checking file versions using file id
+  As a user
+  I want to share file outside of the space
+  So that other users can access the file
+
+
+  Scenario: check the file versions of a shared file
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:7]: You can extend this scenario to have coverage for different share role: See https://github.com/owncloud/ocis/issues/7824#issue-2013952774

**Review by saw-jan — CHANGES_REQUESTED** [review]

**Review by saw-jan — APPROVED** [review]

**Pull request merged**